live-build/misc/migration-scripts/dx_verify

```diff
@@ -74,10 +88,26 @@ function mount_datasets() {
 	touch /var/dlpx-update/"$version"/etc_system_whitelist
 	touch /var/dlpx-update/"$version"/dx_upg_stress_options
 
-	#TODO LX-1771 setup datasets for masking check
+	#
+	# We need a clone of /var/delphix for masking migration scripts.
+	#
+	runningVar=$(mount | awk '/^\/var\/delphix /{ print $3 }')
+	zfs snapshot "$runningVar@$MDS_SNAPNAME" ||
+		die "unable to create snapshot $runningVar@$MDS_SNAPNAME"
+	zfs clone "$runningVar@$MDS_SNAPNAME" "$runningVar/$MDS_SNAPNAME" ||
+		die "unable to create $runningVar/$MDS_SNAPNAME"
+	zfs destroy -d "$runningVar@$MDS_SNAPNAME" ||
+		die "unable to defer destroy $runningVar@$MDS_SNAPNAME"
+	mount -F zfs "$runningVar/$MDS_SNAPNAME" "$root/var/delphix" ||
+		die "unable to mount $root/var/delphix"
+
+	java=/opt/jdk/bin/java
 }
 
 function cleanup_datasets() {
+	if [ -n "$runningVar" ]; then
```

Comment:
Here and below, should we be using a Bash-style conditional (`[[ ]]` ), instead of the posix shell style (`[ ]`)?